utils/src/main/java/org/apache/cloudstack/utils/redfish/RedfishClient.java

```diff
@@ -0,0 +1,423 @@
+
+            System.out.println("\nRequest: " + httpReq);
+
+            HttpClient client;
+            if (ignoreSSL) {
+                client = ignoreSSLCertValidator();
+            } else {// TODO spawn a regular http client
+                throw new RedfishException("Redfish Client does not support SSL certificates. Please set the client to ignore SSL certificates.");
+            }
+            return client.execute(httpReq);
+        } catch (URISyntaxException | IOException | KeyManagementException | NoSuchAlgorithmException e) { // TODO nasty
+                                                                                                          // catch ...
+                                                                                                          // figure it
+                                                                                                          // out if
+                                                                                                          // can find
+                                                                                                          // better
+                                                                                                          // options
+                                                                                                          // latter
+            throw new RedfishException("Failed to execute GET request URL due to exception", e);
+        }
+    }
+
+    /**
+     *
+     */
+    private String getRequestPathForCommand(RedfishCmdType cmd, String resourceId) {
+        switch (cmd) {
+        case GetSystemId:
+            return SYSTEMS_URL_PATH;
+        case GetPowerState:
+            if (StringUtils.isBlank(resourceId)) {
+                throw new RedfishException(String.format("Command '%s' requires a valid resource ID '%s'.", cmd, resourceId));
+            }
+            return String.format("%s%s", SYSTEMS_URL_PATH, resourceId);
+        case ComputerSystemReset:
+            if (StringUtils.isBlank(resourceId)) {
+                throw new RedfishException(String.format("Command '%s' requires a valid resource ID '%s'.", cmd, resourceId));
+            }
+            return String.format("%s%s%s", SYSTEMS_URL_PATH, resourceId, COMPUTER_SYSTEM_RESET_URL_PATH);
+        default:
+            throw new RedfishException(String.format("Redfish client does not support command '%s'.", cmd));
+        }
+    }
+
+    /**
+     *
+     */
+    private String validateAddressAndPrepareForUrl(String hostAddress) {
+        if (NetUtils.isValidIp6(hostAddress)) {
+            return String.format("[%s]", hostAddress);
+        } else if (NetUtils.isValidIp6(hostAddress)) {
+            return hostAddress;
+        } else {
+            throw new RedfishException(String.format("Redfish host address '%s' is not a valid IPv4 or IPv6 address", hostAddress));
```

Comment:
This shouldn't be the case always. Because a hostname should also be supported.

People might have a DNS entry pointing to the iDRAC/IPMI of their server(s).

Reply:
The `InternetDomainName.isValid(hostAddress)` validation should be enought to validate a domain name.

I will be soon committing a few updates,  one of them is a new JUnit test that ensures this method to accept a domain name (e.g. mydomain.org).